enso-org/enso
# Remove optimistic query hooks

### Pull Request Description
- One of many PRs that should improve Dashboard performance.
  - `useMutationState` seems to be taking quite a bit of time (224ms out of 1199ms) in this trace on Firefox:
  - However, I haven't really noticed any real difference, although I haven't tried too hard to find a differene
  - That said, this also vastly simplifies the code in `backendHooks.ts` (by removing the vast majority of the methods)
  - As a downside, well, we will no longer have optimistic updates - but I guess we are preferring a stale-while-revalidate approach anyway (React Query) when it doesn't degrade UX

![image](https://github.com/user-attachments/assets/61899313-7aaf-4158-b382-0a293a97bb71)

![image](https://github.com/user-attachments/assets/f025dd7d-7755-4000-81cd-f57efce4c4bc)

### Important Notes
None

### Testing Instructions
- Theoretically none, as this should only make the logic *more* correct, if anything, because we are removing a layer of indirection.
  - Otherwise, test the features that use the components present in the diff. (assets table, category switcher, user, org, members table and user groups settings pages)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
